allennlp/tests/data/dataset_readers/coreference/coref_reader_test.py

```diff
@@ -184,6 +185,104 @@ def test_read_from_file(self, lazy):
         assert (["they"], 1) in gold_mentions_with_ids
         assert (["the", "clever", "Hong", "Kong", "people"], 1) in gold_mentions_with_ids
 
+    def test_wordpiece_modeling(self):
+        tokenizer = PretrainedTransformerTokenizer("bert-base-cased")
+        conll_reader = ConllCorefReader(
+            max_span_width=self.span_width, wordpiece_modeling_tokenizer=tokenizer
+        )
+        instances = ensure_list(
+            conll_reader.read(str(AllenNlpTestCase.FIXTURES_ROOT / "coref" / "coref.gold_conll"))
+        )
+
+        assert len(instances) == 2
+
+        fields = instances[0].fields
+        text = [x.text for x in fields["text"].tokens]
+
+        assert text == [
+            "[CLS]",
+            "In",
+            "the",
+            "summer",
+            "of",
+            "2005",
+            ",",
+            "a",
+            "picture",
+            "that",
+            "people",
+            "have",
+            "long",
+            "been",
+            "looking",
+            "forward",
+            "to",
+            "started",
+            "emerging",
+            "with",
+            "frequency",
+            "in",
+            "various",
+            "major",
+            "Hong",
+            "Kong",
+            "media",
+            ".",
+            "With",
+            "their",
+            "unique",
+            "charm",
+            ",",
+            "these",
+            "well",
+            "-",
+            "known",
+            "cartoon",
+            "images",
+            "once",
+            "again",
+            "caused",
+            "Hong",
+            "Kong",
+            "to",
+            "be",
+            "a",
+            "focus",
+            "of",
+            "worldwide",
+            "attention",
+            ".",
+            "The",
+            "world",
+            "'",
+            "s",
+            "fifth",
+            "Disney",
+            "park",
+            "will",
+            "soon",
+            "open",
+            "to",
+            "the",
+            "public",
+            "here",
+            ".",
+            "[SEP]",
+        ]
```

Comment:
Could you also include asserts against a few mentions explicitly like the other test case? Ideally in a way that shows how they're different with wordpieces. One issue with the test data currently is that all the words are so common we don't actually see the influence of wordpieces, i.e. no tokens are being split. I'd edit the test data to include some very long words (nonsense is okay) including where the mentions are.